python/cudf_polars/cudf_polars/dsl/expressions/string.py

```diff
@@ -379,6 +385,73 @@ def do_evaluate(
                 ),
                 dtype=self.dtype,
             )
+        elif self.name is StringFunction.Name.SplitN:
+            n = self.options[0]
+            child, expr = self.children
+            column = child.evaluate(df, context=context)
+            if n == 1:
+                plc_column = plc.Column(
+                    self.dtype.plc,
+                    column.obj.size(),
+                    None,
+                    None,
+                    0,
+                    column.obj.offset(),
+                    [column.obj],
```

Comment:
Just to be sure I'm understanding, we're tossing this into the children because Polars expects the output to be of type struct? But if so, how is the dtype working out here if we are constructing the pylibcudf column to be of type `self.dtype.plc` (which presumably is string) and then using `self.dtype` when constructing the cudf-polars `Column` below? Where does the struct come in? Isn't `self.dtype` going to be string?

Reply:
> we're tossing this into the children because Polars expects the output to be of type struct?

Correct

> how is the dtype working out here if we are constructing the pylibcudf column to be of type self.dtype.plc (which presumably is string)

`self.dtype.plc` will be struct here (`self.dtype.polars` will be `struct[string]`). `self.dtype`, probably non-intuitively, of these expression objects are the _result_ type and not the type of the source data.

For context, [here](https://github.com/rapidsai/cudf/blob/branch-25.08/python/cudf_polars/cudf_polars/dsl/translate.py#L152-L154) is where we are getting the result type determined by Polars and then passing it along to our expression objects.